src/Microsoft.Azure.WebJobs.Host/Hosting/ServiceCollectionDecorator.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Azure.WebJobs.Host.Hosting
+{
+    public class ServiceCollectionDecorator : IServiceCollection, ITrackingServiceCollection
+    {
+        public IServiceCollection ServiceCollection { get; set; }
+
+        public List<ServiceDescriptor> Diff { get; private set; } // tracks any service collection modification.
```

Comment:
Same comment on naming (something like `TrackedChanges` would be appropriate here)